test/framework/machinedeployment_helpers.go

```diff
@@ -55,8 +56,8 @@ func CreateMachineDeployment(ctx context.Context, input CreateMachineDeploymentI
 	Expect(input.Creator.Create(ctx, input.InfraMachineTemplate)).To(Succeed())
 }
 
-// GetMachineDeploymentsByClusterInput is the input for GetMachineDeploymentsByCluster.
-type GetMachineDeploymentsByClusterInput struct {
+// GetResourcesByClusterInput is the input for GetMachineDeploymentsByCluster and GetMachinePoolsByCluster.
+type GetResourcesByClusterInput struct {
```

Comment:
This is another breaking change.
However, in this case, IMO we should keep the current type and create a new one for `GetMachinePoolsByCluster`, so we are not coupling the two methods

Reply:
Kept the current type and added a new one for `MachinePools`